apps/board-data-react/src/App.js

```diff
@@ -0,0 +1,140 @@
+import React from "react";
+import mondaySdk from "monday-sdk-js";
+const monday = mondaySdk();
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      context: {},
+      boardIds: [],
+      boards: [],
+      newItemName: ""
+    };
+  }
+
+  componentDidMount() {
+    monday.listen("context", this.getContext);
+  }
+
+  getContext = res => {
+    this.setState({ context: res.data });
+
+    // normalize the data as it can be retrieved from board view or widget that
+    // supports multiple boards
+    let boardIds;
+    if (this.state.context.boardIds) {
+      boardIds = this.state.context.boardIds;
+    }
+    if (this.state.context.boardId) {
+      boardIds = [this.state.context.boardId];
+    }
+
+    this.setState({ boardIds });
+    
+    this.getBoards();
+  };
+
+  getBoards() {
+    const { boardIds } = this.state;
+    if(boardIds.length <= 0) {
+      return;
+    }
+    
+    monday
+      .api(
+        `query {
+      boards(ids: ${boardIds}) {
+        id,
+        name
+
+        items {
+          name,
+          id
+        }
+      }
+    }`
+      )
+      .then(res => {
+        this.setState({ boards: res.data.boards });
+      });
+  }
+
+  addNewItem = event => {
+    event.preventDefault();
+    this.createNewItem(this.state.newItemName, this.state.newItemBoard);
+  };
+
+  createNewItem(title, boardId) {
+    monday
+      .api(
+        `mutation {
+          create_item (board_id: ${boardId}, item_name: "${title}") {
+          name,
+          id
+          }
+          }`
+      )
+      .then(() => {
+        this.getBoards()
+      });
+  }
+
+  changedNewItemName = event => {
+    this.setState({ newItemName: event.target.value });
+  };
+
+  changedNewItemBoard = event => {
+    this.setState({ newItemBoard: event.target.value });
+  };
+
+  render() {
+    const { boards } = this.state;
+
+    let boardOptions = boards.map(board => (
+      <option value={board.id} key={`sb-${board.id}`}>
```

Comment:
what is "sb" stand for?

Reply:
renamed :)

Reply:
changed it to use keys (was using id by mistake at the beginning, thats why the prefix was added)